lib/src/utils/organize_imports/import_collector.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import 'import.dart';
+
+/// A visitor which collects all imports.
+///
+/// Imports are returned in the order they appear in a file when passed to `CompilationUnit.accept`.
+class ImportCollector extends SimpleAstVisitor<List<Import>> {
+  final List<Import> _imports = [];
+
+  @override
+  List<Import> visitImportDirective(ImportDirective node) {
+    _imports.add(Import(node));
+    return null;
+  }
+
+  @override
+  List<Import> visitCompilationUnit(CompilationUnit node) {
+    node.visitChildren(this);
+    return _imports;
+  }
+}
```

Comment:
Any reason not to collect and organize export directives, as well? We'd need to bucket and organize them separately, but otherwise I'd think the logic would be the same.

Reply:
oooh I can look at adding that, I don't think it will be too bad. We just hadn't had as many `#nit export order` code review comments, so it never crossed my mind